device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt

```diff
@@ -86,30 +92,28 @@ class CalendarDelegate : PluginRegistry.RequestPermissionsResultListener {
             return false
         }
 
-        val cachedValues: CalendarMethodsParametersCacheModel? = _cachedParametersMap[requestCode]
-        if (cachedValues == null) {
-            // unlikely scenario where another plugin is potentially using the same request code but it's not one we are tracking so return to
-            // indicate we're not handling the request
-            return false
-        }
+        val cachedValues: CalendarMethodsParametersCacheModel = _cachedParametersMap[requestCode]
+                ?: // unlikely scenario where another plugin is potentially using the same request code but it's not one we are tracking so return to
+                // indicate we're not handling the request
+                return false
 
         when (cachedValues.calendarDelegateMethodCode) {
-            RETRIEVE_CALENDARS_METHOD_CODE -> {
+            RETRIEVE_CALENDARS_REQUEST_CODE -> {
                 return handleRetrieveCalendarsRequest(permissionGranted, cachedValues, requestCode)
```

Comment:
All these handle methods seem to do a check on permissionGranted - can we pull out this check and do it before the when clause?

Reply:
refactored this out now. thanks